<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>TeacupCerberus • TeacupCerberus</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="TeacupCerberus">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">TeacupCerberus</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/TeacupCerberus.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jsilve24/TeacupCerberus">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>TeacupCerberus</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/jsilve24/TeacupCerberus/blob/master/vignettes/TeacupCerberus.Rmd"><code>vignettes/TeacupCerberus.Rmd</code></a></small>
      <div class="hidden name"><code>TeacupCerberus.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co">#library(TeacupCerberus)</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/load_all">load_all</a></span>()</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="co">#&gt; Loading TeacupCerberus</span></a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Random">set.seed</a></span>(<span class="dv">858302</span>)</a></code></pre></div>
<div id="teacup-cerberus-the-model" class="section level1">
<h1 class="hasAnchor">
<a href="#teacup-cerberus-the-model" class="anchor"></a>Teacup Cerberus, The Model</h1>
<p>The Teacup Cerberus Model is very simple. It is essentially just estimating covariation within and between two datasets that have multinomial variation while propogating uncertainty due to count variation into the estimates of covariation. Note: its called Teacup Cerberus because its a simplified version of a model (called Cerberus) that I will eventually put into the <a href="https://jsilve24.github.io/stray/"><em>stray</em> package</a>.</p>
<p>Assume we have two datasets <span class="math inline">\(Y^{(1)}\)</span> (e.g., 16S Microbiome sequencing) and <span class="math inline">\(Y^{(2)}\)</span> (e.g., paired bulk RNA-seq samples). We measure counts but want to look at correlations between log-ratios (in particular CLR coordinates). We therefore have uncertainty in our estimates of relative abundances (and therfore log-ratios) due to count variation. We can model this uncertainty using a simple multinomial dirichlet model. <span class="math display">\[
\begin{align}
Y_i^{(1)} &amp;\sim \text{Multinomial}(\pi^{(1)}_i) \\
Y_i^{(2)} &amp;\sim \text{Multinomial}(\pi^{(2)}_i) \\
\pi^{(1)}_i &amp;\sim \text{Dirichlet}(\alpha^{(1)}) \\
\pi^{(2)}_i &amp;\sim \text{Dirichlet}(\alpha^{(2)}).
\end{align}
\]</span> Which has a posterior of <span class="math display">\[
\begin{align}
\pi^{(1)}_i | Y^{(1)}_i &amp;\sim \text{Dirichlet}(\alpha^{(1)} + Y^{(1)}_i) \\
\pi^{(2)}_i | Y_i^{(2)} &amp;\sim \text{Dirichlet}(\alpha^{(2)}+Y^{(2)}_i).
\end{align}
\]</span> We can therefore draw posterior samples directly from the posterior of this model. Denote <span class="math inline">\(s\)</span>-th posterior samples of the <span class="math inline">\(i\)</span>-th observation as <span class="math inline">\(\pi^{(1)s}_i\)</span> and <span class="math inline">\(\pi^{(2)s}_i\)</span> respectively. If we collect enough samples from the posterior we can obtain an estimate of our uncertainty, each posterior sample representing one posible true value of the relative abundances <span class="math inline">\(\pi^{(1)}_i\)</span> and <span class="math inline">\(\pi^{(2)}_i\)</span>. For each posterior sample we can then calculate the correlation within and between the two datasets in the CLR coordinates as follows: <span class="math display">\[
\begin{align}
\eta^{(1)s}_i &amp;= \text{CLR}(\pi^{(1)s}_i) \\
\eta^{(2)s}_i &amp;= \text{CLR}(\pi^{(2)s}_i) \\
\Sigma_s &amp;= \text{corr}\left(\begin{bmatrix} \eta^{(1)s} \\ \eta^{(2)s} \end{bmatrix}\right)
\end{align}
\]</span> We then have posterior samples of the correlation matrix, each sample <span class="math inline">\(\Sigma^s\)</span> has the following form: <span class="math display">\[
\Sigma_s = \begin{bmatrix} \Sigma_s^{(1,1)} &amp; \Sigma_s^{(1,2)} \\
 \Sigma_s^{(2,1)} &amp; \Sigma_s^{(2,2)}
\end{bmatrix}
\]</span> where <span class="math inline">\(\Sigma_s^{(1,1)}\)</span> represents the correlation between CLR coordinates in the first dataset, <span class="math inline">\(\Sigma_s^{(2,2)}\)</span> represents the correlation between CLR coordinates in the second dataset, and <span class="math inline">\(\Sigma_s^{(1,2)}\)</span> represents correlations cross correlations between CLR coordinates in the two datasets.</p>
<p>These posterior samples can then be investigated and summarised (e.g., for each element of the matrix <span class="math inline">\(\Sigma\)</span>) the posterior mean and quantiles can be summarized across the samples. We can also easily compute other statistics e.g., covariation in other Log-ratio coordinates, proportionality, or the IQLR transformed covariance. Below I will go over a simple example of how to do these things using simulated data.</p>
</div>
<div id="example-usage" class="section level1">
<h1 class="hasAnchor">
<a href="#example-usage" class="anchor"></a>Example Usage</h1>
<p>First lets simulate some random data. Lets say there are 10 “parts” (e.g., taxa) in the first dataset, 15 parts (e.g., genes) in the second and a total of 12 samples in each. Note it is important that these samples are paired, e.g., one sample in the first dataset corresponds to one sample in the second.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">D1 &lt;-<span class="st"> </span><span class="dv">10</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2">D2 &lt;-<span class="st"> </span><span class="dv">15</span></a>
<a class="sourceLine" id="cb2-3" data-line-number="3">N &lt;-<span class="st"> </span><span class="dv">12</span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4">sim &lt;-<span class="st"> </span><span class="kw"><a href="../reference/cerberus_sim.html">cerberus_sim</a></span>(<span class="dt">D1=</span>D1, <span class="dt">D2=</span>D2, <span class="dt">N=</span>N)</a></code></pre></div>
<p>Lets look at the data we simulated:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">sim<span class="op">$</span>Y1 <span class="co"># the first dataset</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="co">#&gt;       s1   s2   s3   s4   s5   s6   s7   s8   s9  s10  s11  s12</span></a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="co">#&gt; c1   938   58   15 1982   39  228  293   55   86  352 2281   16</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="co">#&gt; c2  3347   33   86    0   72   79  587    6    8  762  173   58</span></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co">#&gt; c3    53    9  411   55  131  224   19 5260   26  175  313 1369</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="co">#&gt; c4   332 5857 5858 6921 4774 6106 1192 2503 5843  585  782  684</span></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="co">#&gt; c5   169   15   22   50 1015    1  301   83    2  427  172  478</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="co">#&gt; c6   588   25   10   11  936  109 3467   85  105 3941  637 5827</span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="co">#&gt; c7   129   36   57    8   26   10   15  613    2  194  309    4</span></a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="co">#&gt; c8  1167  775   16    3  175   46  372   49    0  133 2403   26</span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="co">#&gt; c9   675   29    8    5   12   39    6   28    1  506   66  322</span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="co">#&gt; c10  238   16   14   21  436   27  139  156    6  838  314  196</span></a>
<a class="sourceLine" id="cb3-13" data-line-number="13">sim<span class="op">$</span>Y2 <span class="co"># the second dataset</span></a>
<a class="sourceLine" id="cb3-14" data-line-number="14"><span class="co">#&gt;       s1   s2   s3   s4   s5   s6   s7   s8   s9  s10  s11  s12</span></a>
<a class="sourceLine" id="cb3-15" data-line-number="15"><span class="co">#&gt; d1    11    0  362    4    0    0   62    0   49   88    0    0</span></a>
<a class="sourceLine" id="cb3-16" data-line-number="16"><span class="co">#&gt; d2     3    0  447   78  130    1  182    7    0  484    0    1</span></a>
<a class="sourceLine" id="cb3-17" data-line-number="17"><span class="co">#&gt; d3   216    0    4   22  304    0   34    5    2  683    0    0</span></a>
<a class="sourceLine" id="cb3-18" data-line-number="18"><span class="co">#&gt; d4    38    1 1074    4   47    0   43   56   11   90    0    0</span></a>
<a class="sourceLine" id="cb3-19" data-line-number="19"><span class="co">#&gt; d5   169    1 1029    5  354    0   84   55   24   29    0    0</span></a>
<a class="sourceLine" id="cb3-20" data-line-number="20"><span class="co">#&gt; d6   791    1  162   36  825    0   82   20    5   87    0    6</span></a>
<a class="sourceLine" id="cb3-21" data-line-number="21"><span class="co">#&gt; d7    31    0 1371    0  260    0    2    0    8    0    0    0</span></a>
<a class="sourceLine" id="cb3-22" data-line-number="22"><span class="co">#&gt; d8    39    0   49    0  183    0   34   73   32   17    0    0</span></a>
<a class="sourceLine" id="cb3-23" data-line-number="23"><span class="co">#&gt; d9     0    0   22   14   19    0   70    1    0    2    0    0</span></a>
<a class="sourceLine" id="cb3-24" data-line-number="24"><span class="co">#&gt; d10   17  236  791   24  785    0  560 1299   10 1297    0    2</span></a>
<a class="sourceLine" id="cb3-25" data-line-number="25"><span class="co">#&gt; d11  294    3  605    4  869    6   35  271   61  257    0    0</span></a>
<a class="sourceLine" id="cb3-26" data-line-number="26"><span class="co">#&gt; d12    2    0  260    5   26    0   82  338    1  211    0    0</span></a>
<a class="sourceLine" id="cb3-27" data-line-number="27"><span class="co">#&gt; d13  171    0   52    1    2    0   30    5   70    2    0    0</span></a>
<a class="sourceLine" id="cb3-28" data-line-number="28"><span class="co">#&gt; d14 6265 7833  591 9617 1775 5993 7454 4795 7631 5024 7740 9960</span></a>
<a class="sourceLine" id="cb3-29" data-line-number="29"><span class="co">#&gt; d15   28    0  767    0   74    0   66   12    7   56    0    1</span></a></code></pre></div>
<p>Note that the model assumes that the <span class="math inline">\(i\)</span>-th row of <code>Y1</code> is paired with the <span class="math inline">\(i\)</span>-th row of <code>Y2</code>.</p>
<p>It is very easy to fit the Cerberus Teacup model.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="co"># Choose priors</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2">alpha1 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(.<span class="dv">5</span>, D1)</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">alpha2 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/rep">rep</a></span>(.<span class="dv">5</span>, D2)</a>
<a class="sourceLine" id="cb4-4" data-line-number="4"></a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="co"># Fit Model</span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6">Sigma &lt;-<span class="st"> </span><span class="kw"><a href="../reference/teacup_cerberus.html">teacup_cerberus</a></span>(sim<span class="op">$</span>Y1, sim<span class="op">$</span>Y2, alpha1, alpha2)</a></code></pre></div>
<p>We can now look at inferred covariation between CLR coordinates. As a convenience, the function <code>teacup_cerberus</code> tries to use rownames in <code>Y1</code> and <code>Y2</code> to allow us to easily index within the returned <code>Sigma</code>. We can look at the posterior distribution of the covariance between part <code>c1</code> in <code>Y1</code> and <code>d2</code> in <code>Y2</code> as follows:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/density">density</a></span>(Sigma[<span class="st">"c5"</span>,<span class="st">"d2"</span>,]))</a></code></pre></div>
<p><img src="TeacupCerberus_files/figure-html/unnamed-chunk-5-1.png" width="384" style="display: block; margin: auto;"> So we see that most of the mass of the posterior is centered on positive values of <span class="math inline">\(Cov(c_1, d_2)\)</span> suggesting the CLR cooridnates corresponding to these two things are positively correlated.</p>
<p>There are a number of different helpful representations that users may want to look at.</p>
<div id="extracting-correlations-from-sigma" class="section level2">
<h2 class="hasAnchor">
<a href="#extracting-correlations-from-sigma" class="anchor"></a>Extracting Correlations from Sigma</h2>
<p>If you want to view the posterior over correlations rather than covariances just use the function <code>sigma2cor</code>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">Sigma.cor &lt;-<span class="st"> </span><span class="kw"><a href="../reference/sigma2cor.html">sigma2cor</a></span>(Sigma)</a></code></pre></div>
</div>
<div id="calculate-proportionality-rather-than-covariance" class="section level2">
<h2 class="hasAnchor">
<a href="#calculate-proportionality-rather-than-covariance" class="anchor"></a>Calculate Proportionality rather than Covariance</h2>
<p>Proportionality is a nice alternative to correlation/covariance when working with compositional data. If two CLR coordinates are proportional then it can be concluded that the taxa/genes represented by those CLR coordinates are infact correlated. This is not true when using normal correlation however. A good review of proportionality can be found in <span class="citation">Lovell et al. (2015)</span>. <span class="citation">Lovell et al. (2015)</span> proposed <span class="math inline">\(\phi\)</span> statistics as a measure of proportionality. The lower the value of <span class="math inline">\(\phi\)</span> the more proportional two things are. <span class="math inline">\(\phi\)</span> statistics can be calculated directly from <code>Sigma</code> using the function <code><a href="https://www.rdocumentation.org/packages/RcppCoDA/topics/clrvar2phi">RcppCoDA::clrvar2phi</a></code> as below</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">Sigma.phi &lt;-<span class="st"> </span>RcppCoDA<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/RcppCoDA/topics/clrvar2phi">clrvar2phi</a></span>(Sigma)</a></code></pre></div>
</div>
<div id="represent-sigma-in-iqlr-rather-than-clr" class="section level2">
<h2 class="hasAnchor">
<a href="#represent-sigma-in-iqlr-rather-than-clr" class="anchor"></a>Represent Sigma in IQLR rather than CLR</h2>
<p>The IQLR transform <span class="citation">(Wu et al. 2017)</span> is a nice alternative for the CLR that deal with compositional asymetry. It’s a bit of a complicated transform, and I am not going to go into details here. What I will say is that when working with Microbiome data, the IQLR mahy be helpful when you want to pretend like your acctually measuring correlation between taxa rather than between “log-ratios”.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">Sigma.iqlr &lt;-<span class="st"> </span>RcppCoDA<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/RcppCoDA/topics/iqlr">clrvar2iqlrvar</a></span>(Sigma)</a></code></pre></div>
</div>
</div>
<div id="ranking-hits" class="section level1">
<h1 class="hasAnchor">
<a href="#ranking-hits" class="anchor"></a>Ranking “hits”</h1>
<p>I imagine that many who use this package will be doing so to find “hits” parts in <code>Y1</code> that are correlated to parts in <code>Y2</code>. <code>teacup_cerberus</code> outputs a posterior over covaraince matricies. So how do you get hits? Ideally we want to find those “hits” that correspond to cases where we are not only highly certainty that there is a non-zero covariance but also that the covariance is not neglibly non-zero; e.g., we want cases were we are highly certain the covariation is large. I like to use a statistic that I call ECDF(0) for this purpose. It essentially calculate the quantile corresponding to zero covariation from the empirical cumulative distribution function formed by posterior samples. Don’t worry if that doesn’t make sense. Essentially values of ECDF(0) close to 1 give suggest high certainty that the covariation is positive and values close to 0 give high certainty that the covariation is negative. Values near 0.5 refelct either that we have high certainty that there is no covariation or we are just extreemly uncertain regarding what the true covariation is. To make ECDF(0) useful for users of this package I have written the function <code>filter_sigma_ecdf</code>. Essentially you give a value of <code>lambda</code> and the function only returns hits where <code>ECDF(0)&gt;=lambda</code> or <code>ECDF(0) &lt;= 1-lambda</code>. It also ranks the output so the first rows correspond to the things we are most confident are large (positive or negative).</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw"><a href="../reference/filter_sigma_ecdf.html">filter_sigma_ecdf</a></span>(Sigma, <span class="fl">0.95</span>, <span class="dt">use_names=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="co">#&gt; # A tibble: 186 x 3</span></a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="co">#&gt; # Groups:   Coord1 [25]</span></a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="co">#&gt;    Coord1 Coord2 ECDF0</span></a>
<a class="sourceLine" id="cb9-5" data-line-number="5"><span class="co">#&gt;    &lt;chr&gt;  &lt;chr&gt;  &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb9-6" data-line-number="6"><span class="co">#&gt;  1 c1     c4         0</span></a>
<a class="sourceLine" id="cb9-7" data-line-number="7"><span class="co">#&gt;  2 c1     d10        1</span></a>
<a class="sourceLine" id="cb9-8" data-line-number="8"><span class="co">#&gt;  3 c1     d14        0</span></a>
<a class="sourceLine" id="cb9-9" data-line-number="9"><span class="co">#&gt;  4 c10    c5         0</span></a>
<a class="sourceLine" id="cb9-10" data-line-number="10"><span class="co">#&gt;  5 c10    c6         0</span></a>
<a class="sourceLine" id="cb9-11" data-line-number="11"><span class="co">#&gt;  6 c10    d3         0</span></a>
<a class="sourceLine" id="cb9-12" data-line-number="12"><span class="co">#&gt;  7 c2     c3         1</span></a>
<a class="sourceLine" id="cb9-13" data-line-number="13"><span class="co">#&gt;  8 c2     c4         1</span></a>
<a class="sourceLine" id="cb9-14" data-line-number="14"><span class="co">#&gt;  9 c2     c5         1</span></a>
<a class="sourceLine" id="cb9-15" data-line-number="15"><span class="co">#&gt; 10 c2     c6         0</span></a>
<a class="sourceLine" id="cb9-16" data-line-number="16"><span class="co">#&gt; # … with 176 more rows</span></a></code></pre></div>
<p>This is just an example of one way of interpreting / using output from <code>teacup_cerberus</code>. Note this filtering method would work well for <code>Sigma.iqlr</code> and <code>Sigma</code> (the later is in the CLR) because there the things that are “interesting” will have a posterior not centered about zero therefore we are looking for posteriors far from zero. However, for <code>Sigma.phi</code> we need a different tact because there we are just looking for hits that have values close to zero. Alternative approaches and summarizations can be easily chosen to summarize phi statistics (for example, look for those “hits” with the lowest posterior mean etc…).</p>
</div>
<div id="picking-between-clr-iqlr-and-phi" class="section level1">
<h1 class="hasAnchor">
<a href="#picking-between-clr-iqlr-and-phi" class="anchor"></a>Picking between CLR, IQLR, and Phi</h1>
<p>The CLR, IQLR, and Phi statistics are really 3 ways of viewing the same posterior samples. I realize I have given users quite a few options here so here is some guidance. Lets say that you are looking for “hits” as I described above. I would likely start by looking at the Phi statistics. Any relationship between two parts (e.g., Phi close to zero) is strong evidence that the two parts are truely correlated in absolute abundance! Not just relative abundance! This is a huge advantage. So if I found some really strong signal with Phi statistics I would be most confident about the interpretation of my data. That siad, Phi statistics can miss true correlation because they are only looking for proportionality (which is a subset of all correlations that is valid with compositional data). So if I didn’t find anything with Phi statistics I would likely look with the IQLR. My personal experience and that of a few benchmarking studies has suggested that IQLR is better than CLR for things like this (where you want base scale inference). That said, because the IQLR lacks the theoretical underpinning I would use it with some skepticism. Last I would try the CLR and would have the most skepticism about those results.</p>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-lovell2015">
<p>Lovell, David, Vera Pawlowsky-Glahn, Juan José Egozcue, Samuel Marguerat, and Jürg Bähler. 2015. “Proportionality: A Valid Alternative to Correlation for Relative Data.” <em>PLoS Computational Biology</em> 11 (3). Public Library of Science: e1004075.</p>
</div>
<div id="ref-wu2017">
<p>Wu, Jia R, Jean M Macklaim, Briana L Genge, and Gregory B Gloor. 2017. “Finding the Centre: Corrections for Asymmetry in High-Throughput Sequencing Datasets.” <em>arXiv Preprint arXiv:1704.01841</em>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#teacup-cerberus-the-model">Teacup Cerberus, The Model</a></li>
      <li>
<a href="#example-usage">Example Usage</a><ul class="nav nav-pills nav-stacked">
<li><a href="#extracting-correlations-from-sigma">Extracting Correlations from Sigma</a></li>
      <li><a href="#calculate-proportionality-rather-than-covariance">Calculate Proportionality rather than Covariance</a></li>
      <li><a href="#represent-sigma-in-iqlr-rather-than-clr">Represent Sigma in IQLR rather than CLR</a></li>
      </ul>
</li>
      <li><a href="#ranking-hits">Ranking “hits”</a></li>
      <li><a href="#picking-between-clr-iqlr-and-phi">Picking between CLR, IQLR, and Phi</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Justin Silverman.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
